### DISCUSSION TITLE
Anyof , specific required fields

### DISCUSSION THREAD

**Discussion by Meriyemelhajoui** [microsoft/typespec#3848] [Json Schema] [answered]
Hello , While defining my schema for JSON Document , using typespec , I was wondering how I can present this : 
**one of these two properties is required** : name of type string and address of type string. 
 ```
      model Person {
// OneOf the two properties is required ??
 name: string | null; 
 address : string | null;
     ...AdditionalProperties<never>;
}

```

**Comment by timotheeguerin** [comment]
Currently the only option is to use a union to define both combination

**Comment by Meriyemelhajoui** [comment]
so I need to define the name as separate modele and also the address or what exactly ? the name can be required or the address one of the Two

**Reply by timotheeguerin** [reply, marked as answer]
Yeah something like that
```
model Person {
   
}

model PersonWIthName {
  name: string;
  address?: string;
}

model PersonWithAddress {
  name?: string;
  address: string;
}

union Person {
  PersonWIthName, PersonWithAddress
}
```

**Comment by Meriyemelhajoui** [comment]
Yeah I see , thank you !
